### PR TITLE
Fix decoding of From field value

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
Decode From field value as specified by the MMS standard (encoded string value instead of text string). Fixes

https://bugs.launchpad.net/ubuntu/+source/ofono/+bug/1360403

### CODE REVIEW

File: mms/decoder.go [hunk 1]
```diff
@@ -386,6 +386,7 @@ func (dec *MMSDecoder) Decode(pdu MMSReader) (err error) {
 		case FROM:
 			dec.Offset++
 			size := int(dec.Data[dec.Offset])
+			val_start := dec.Offset
```
Comment (sergiusens) [inline mms/decoder.go]: this is not a valid go var name, should be valStart

File: mms/decoder.go [hunk 2]
```diff
@@ -393,11 +394,10 @@ func (dec *MMSDecoder) Decode(pdu MMSReader) (err error) {
 				break
 			case TOKEN_ADDRESS_PRESENT:
 				// TODO add check for /TYPE=PLMN
-				var from string
-				from, err = dec.ReadString(&reflectedPdu, "From")
-				// size - 2 == size - token - '0'
-				if len(from) != size-2 {
-					err = fmt.Errorf("From field is %d but expected size is %d", len(from), size-2)
+				_, err = dec.ReadEncodedString(&reflectedPdu, "From")
```
Comment (sergiusens) [inline mms/decoder.go]: oh darn, I didn't add up the pieces correctly

File: cmd/nuntium/mediator.go [hunk 3]
```diff
@@ -186,7 +186,11 @@ func (mediator *Mediator) getMRetrieveConf(mNotificationInd *mms.MNotificationIn
 		log.Print("Download issues: ", err)
 		return
 	} else {
-		storage.UpdateDownloaded(mNotificationInd.UUID, filePath)
+		err := storage.UpdateDownloaded(mNotificationInd.UUID, filePath)
```
Comment (sergiusens) [inline cmd/nuntium/mediator.go]: nice catch, can I recomment:

if err := ....; err != nil {
    log....
    return
}

to contain that err var?

File: mms/decoder.go (hunk 1 above)
Comment (jhodapp) [inline mms/decoder.go]: Is it customary in all Go code to use camel-case?

Comment (alfonsosanchezbeato) [inline mms/decoder.go]: Too used to C... I'll fix that

**Comment by alfonsosanchezbeato** [comment]
Comments addressed + new test

**Pull request merged**